classes/mail/variables/SubmissionEmailVariable.php

```diff
@@ -29,7 +29,7 @@ protected function getSubmissionPublishedUrl(Context $context): string
             $context->getPath(),
             'catalog',
             'book',
-            $this->submission->getBestId()
+            $this->submission->getCurrentPublication()->getData('urlPath') ?? $this->submission->getId()
```

Comment:
If the `urlPath` is rarely used out of this URL generation context, then I think it's better to keep this extra detail encapsulated for now (just to make it easier to do replaces later) :)
About the others, you already know my opinion about the `getData`/`setData()`, but as we're moving to the Laravel model, nothing against dropping them.

Reply:
> then I think it's better to keep this extra detail encapsulated for now

Agreed, I hated these changes as I was making them and the situation didn't improve over the weekend. I'm planning to revert this part.